src/realm/realmActions.js

```diff
@@ -0,0 +1,38 @@
+import { homeNarrow, specialNarrow } from '../utils/narrow';
+import { tryUntilSuccessful } from '../utils/async';
+import { getSubscriptions, getMessages, getStreams, getUsers } from '../api';
+import { messageFetchSuccess } from '../message-list/messagesActions';
+import { initSubscriptions } from '../subscriptions/subscriptionsActions';
+import { initStreams } from '../streamlist/streamsActions';
+import { initUsers } from '../users/userListActions';
+import { FETCH_INITIAL_REALM_DATA } from '../constants';
+
+export const fetchInitialRealmData = () => ({
+  type: FETCH_INITIAL_REALM_DATA,
+});
+
+export const fetchEssentialInitialData = (auth) =>
+  async (dispatch) => {
+    const [subscriptions, messages] = await Promise.all([
+      await tryUntilSuccessful(() => getSubscriptions(auth)),
```

Comment:
Once the `register` call is optimized I think we should merge these into just one call to `register`. It looks really janky if all the data isn't loaded but the user can still interact with the app